atomicfu/src/concurrentTest/kotlin/kotlinx/atomicfu/parking/TimeArithmeticTests.kt

```diff
@@ -0,0 +1,47 @@
+package kotlinx.atomicfu.parking
+
+import kotlin.random.Random
+import kotlin.test.Test
+import kotlin.test.assertTrue
+
+class TimeArithmeticTests {
+    
+    
+    
+    @Test
+    fun timeArithmeticTest() {
+        val currentTimes = listOf<Long>(
+            31_560_000, // one year in seconds
+            31_560_000 * 5,
+            31_560_000 * 25,
+            31_560_000 * 60,
+            31_560_000 * 100,
+            31_560_000 * 500,
+            31_560_000 * 1000,
+            31_560_000 * 10000,
+            )
+        
+        currentTimes.forEach { currentTimeInSeconds -> 
+            
+            
+            // Test Long
+            repeat(1000) {
+                val nanos = Random.nextLong()
+                val updatedTime = currentTimeInSeconds.addNanosToSeconds(nanos)
+                if (nanos > 0) assertTrue { updatedTime > currentTimeInSeconds }
+                if (nanos < 0) assertTrue { updatedTime < currentTimeInSeconds }
+            }
+
+            // Test Int
+            repeat(1000) {
+                val currentTimeInInt = currentTimeInSeconds.toInt()
+                val nanos = Random.nextLong()
+                val updatedTime = currentTimeInInt.addNanosToSeconds(nanos)
+                if (nanos > 0) assertTrue { updatedTime > currentTimeInSeconds && updatedTime <= Int.MAX_VALUE }
```

Comment:
A stronger version of the check: if the value does not clamp, the `(updatedTime - currentTime) == nanos / 1_000_000_000`; if the value clamps, it's because as a `Long` value, it would exceed an `Int`. This describes what we want from `addNanosToSeconds` more precisely. As is, when I write `internal fun Long.addNanosToSeconds(nanos: Long): Long = this + nanos / 2_000_000_000`, for example, no tests fail.